llvm/include/llvm/Passes/StandardInstrumentations.h

```diff
@@ -46,6 +46,24 @@ class PrintIRInstrumentation {
   void registerCallbacks(PassInstrumentationCallbacks &PIC);
 
 private:
+  enum class SuffixType {
+    Before,
+    After,
+    Invalidated,
+  };
+
+  struct PassRunDescriptor {
+    const Module *M;
+    SmallString<128> DumpIRFilename;
```

Comment:
const?

Reply:
I did this because I use structural binding to unpack the struct, and there doesn't seem a way to have the unpacked members lose the const qualifier. I need to modify DumpIRFilename to append the suffix and I don't want to copy it unnecessarily.

```cpp
auto [M, DumpIRFilename, IRName, StoredPassID] = popPassRunDescriptor(PassID);
```

I'll just unpack the struct normally -- perhaps I got a little too excited about using structural binding.

Reply:
I'll admit structured binding looks a lot nicer, this is fine. I wouldn't worry too much about copies in this code though, it shouldn't be performance critical.

Reply:
Just made the copy